include/swift/AST/DiagnosticsParse.def

```diff
@@ -619,8 +619,9 @@ ERROR(expected_type_function_result,PointsToFirstBadToken,
       "expected type for function result", ())
 ERROR(generic_non_function,PointsToFirstBadToken,
       "only syntactic function types can be generic", ())
-ERROR(rethrowing_function_type,PointsToFirstBadToken,
-      "only function declarations may be marked 'rethrows'", ())
+ERROR(rethrowing_function_type,none,
+      "only function declarations may be marked 'rethrows'; "
+      "did you mean 'throws'?", ())
```

Comment:
Summoning @rjmccall. A `rethrows` function type is useful, but it also makes the language more complicated, and I thought we decided not to do it.